MainWindow.xaml.cs

```diff
@@ -270,5 +268,54 @@ private void ButtonFill_Click(object sender, RoutedEventArgs e)
         {
             PanZoom.ViewModel.FillRegion();
         }
+
+        #region Closing
+
+        public bool UnsavedChanges
```

Comment:
This property seems odd to me.

I'd assume we *always* want to listen to the `CollectionChanged` events.  And anytime a change occurs, we set `unsavedChanges` to true.  Why are we removing the `ChangesObserved` event if we set to true?

Reply:
we could do that, I just prefer not to have requests coming in all the time when they are not necessary.

Reply:
Ah, I was typing out the steps to ask a question about it and now I see how it works.  Clever.